src/OpenVocabularyAssociationStorage.php

```diff
@@ -160,6 +161,18 @@ protected function doPostSave(EntityInterface $entity, $update) {
   protected function doDelete($entities) {
     parent::doDelete($entities);
 
+    // Invalidate cache for all entity types affected.
+    // We need to do this to guarantee entity lists are
+    // correctly expired when associations are removed.
+    foreach ($entities as $entity) {
+      $fields = $entity->getFields();
+      foreach ($fields as $field) {
+        [$entity_type_id, $bundle, $field_id] = explode('.', $field);
+        $entity_type = \Drupal::entityTypeManager()->getDefinition($entity_type_id);
+        Cache::invalidateTags($entity_type->getListCacheTags());
```

Comment:
Don't we need to update `/node/nid` and `/jsonapi/node/[bundle]/[uid]` pages? These pages don't contain list tags